networkx/generators/tests/test_random_graphs.py

```diff
@@ -247,6 +247,29 @@ def test_gnp(self):
                 edges += sum(1 for _ in generator(10, 0.99999, directed=True).edges())
             assert abs(edges / float(runs) - 90) <= runs * 2.0 / 100
 
+            # assert that edges are generated with correct probability
+            runs = 5000
+            n = 5
+            for p in [0.2, 0.8]:
+                for directed in [False, True]:
+                    edge_counts = [[0] * 5 for row in range(5)]
+                    for i in range(runs):
+                        G = nx.generators.random_graphs.fast_gnp_random_graph(
+                            n, p, directed=directed
+                        )
+                        for (v, w) in G.edges:
+                            edge_counts[v][w] += 1
+                            if not directed:
+                                edge_counts[w][v] += 1
+                    for v in range(n):
+                        for w in range(n):
+                            if v == w:
+                                # There should be no loops
+                                assert edge_counts[v][w] == 0
+                            else:
+                                # Each edge should have been generated with probability close to p
+                                assert abs(edge_counts[v][w] / float(runs) - p) <= 0.03
```

Comment:
As currently defined, this will run 4 times since it is inside the for-loop over random graph generators. The easiest way to fix this would be to move it into it's own test function (or method in this case).

Reply:
Thanks for spotting that. I had intended to run the test for all four generators, but using a specific generator on line 257 was definitely a mistake. I've fixed it now so it uses the loop variable `generator` as the function.

Hopefully this is okay? If you think it's a waste of time testing the simpler generators I can move it outside the loop.